ExploreASL_ImportMaster.m

```diff
@@ -19,41 +19,77 @@
 
     %% Import Workflow
     
-    % We expect x.DataParPath to be the sourceStructure.json file, so the root directory should be the study directory
-	[x.StudyRoot,~,~] = xASL_fileparts(x.DataParPath);
+    % We expect x.opts.StudyRoot to be the sourceStructure.json file, so the root directory should be the study directory
+    if isfield(x, 'dir')
```

Comment:
Wouldn't be good practice to check if a field exists and using it in the second part of the statement, right?
Because if it does not exist, then isempty(x.dir.StudyRoot) should lead to an error or not?

Reply:
In Matlab && is not evaluated completely if one part is already false. A && B doesn't try to evaluate the condition B if A is already false.

So you can use 
`if isfield(x, 'dir') && isfield(x.dir, 'StudyRoot') && ~isempty(x.dir.StudyRoot)`

And it won't crash if 'StudyRoot' field is missing because it will never reach the third condition.
That's just a general comment to answer Michaels question - haven't checked Henk's proposition in detail because there might be missing `~` before `isempty`

Reply:
Ah okay, sure, then Henk's proposition should work... but guys, we're just making one conditional statement out of two. That will make the code faster by like 0.000001%. I don't think that's really worth a discussion or a change request on an open pr.

Reply:
yes - good to keep the code clean and short, but indeed a tiny issue...